src/components/skills-quiz/SkillsQuizStepper.jsx

```diff
@@ -130,11 +132,17 @@ const SkillsQuizStepper = () => {
     );
   }, []);
 
+  useEffect(() => {
+    if (goal === DROPDOWN_OPTION_IMPROVE_CURRENT_ROLE) {
+      setIsStudentChecked(false);
+    }
+  }, [goal]);
+
   // will be true if goal or skills changed not because of first render, if link shared and there are more than one
   // selected skills, or if skillsVisible variable is ever been true for once.
   const skillsVisible = useMemo(() => (!isFirstRender || skillsVisible || (selectedSkills?.length > 0)),
     [goal, selectedSkills]);
-  const jobsDropdownsVisible = useMemo(() => !isFirstRender, [skills]);
+  const jobsDropdownsVisible = useMemo(() => !isFirstRender || skills?.length > 0, [skills]);
```

Comment:
What's the reason behind adding `skills?.length > 0`

Reply:
it was because of issues of some lifecycle things i was facing with tests, it was a safe but redundant statement, so i have found a way to make my tests work and modified them. reverted this statement,

Reply:
yes it will automatically be disabled and unchecked to avoid confusion.